libcontainer/mount_linux.go

```diff
@@ -26,8 +27,8 @@ func (e *mountError) Error() string {
 	} else {
 		out += e.target
 	}
-	if e.procfd != "" {
-		out += " (via " + e.procfd + ")"
+	if e.srcFD != "" || e.dstFD != "" {
+		out += " (via " + e.srcFD + ":" + e.dstFD + ")"
```

Comment:
Isn't this line confusing if `e.srcFD` is empty?

Reply:
I think it's understandable enough (`via :/proc/self/fd/5`) yet simple.

Before:
```
mount /proc/self/fd/11:/sys/fs/cgroup/systemd (via /proc/self/fd/12), flags: 0x20502f: operation not permitted
```

After:
```
mount /proc/self/fd/11:/sys/fs/cgroup/systemd (via :/proc/self/fd/12), flags: 0x20502f: operation not permitted
```

Or would you prefer something like this (untested):
```go
if e.srcFD != "" || e.dstFD != "" {
	const none="<none>"
	s := e.srcFD
	if s == "" {
		s = none
	}
	d := e.dstFD
	if d = "" {
		d  = none
	}
	out += " (via " + s + ":" + d + ")"
```

Reply:
If you can think of a better way to represent the error, let me know, I tried a few alternatives and did not like them.

Reply:
Heh, yeah, not sure I like other alternatives I'm thinking either. What the PR currently does is good enough for me, thanks! :)

Reply:
Redid it again, see https://github.com/opencontainers/runc/pull/3512#discussion_r921532761